### PR TITLE
[#3948] Update dotnet samples to target .Net 8 - Essential samples

### PR DESCRIPTION
**Pull request by andres-robinet-sw**
Addresses # 3948

## Description
This PR updates the Essential samples to target .NET 8

## Specific Changes
- Updated the following samples to target .NET 8
  - 02.echo-bot
  - 03.welcome-user
  - 05.multi-turn-promp
  - 06.using-cards
  - 07.using-adaptive-cards
  - 08.suggested-actions
  - 13.core-bot
  - 15.handling-attachments
  - 40.timex-resolution
  - 43.complex-dialog
  - 45.state-management

## Testing
These images show the updated bots working as expected.
06.using-cards
echo-bot
03.welcome-user
welcome-user
05.multi-turn-promp
multi-turn-prompt
06.using-cards
using-cards
07.using-adaptive-cards
using-adaptive-cards
08.suggested-actions
suggested-actions
13.core-bot
core-bot-dotnet
15.handling-attachments
handling-attachments
handling-attachments-2
handling-attachments-3
40.timex-resolution
timex-resolution
43.complex-dialog
complex-dialog
complex-dialog-2
45.state-management
state-management

### CODE REVIEW

**Review by JhontSouth — APPROVED** [review]

File: samples/csharp_dotnetcore/07.using-adaptive-cards/AdaptiveCardsBot.csproj [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AdaptiveCards" Version="2.7.3" />
+    <PackageReference Include="AdaptiveCards" Version="3.1.0" />
```
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/07.using-adaptive-cards/AdaptiveCardsBot.csproj]: Im not sure about updating from 2.x to 3.x. But if the bot is working, why not.

File: samples/csharp_dotnetcore/40.timex-resolution/README.md
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/40.timex-resolution/README.md:1]: We tend to not change other things that are not related to updating BB and .NET versions, so it is easier to review and its focused on the associated issue.

**Review by sw-joelmut — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]
:shipit:
